cpp/include/cuspatial/experimental/type_utils.hpp

```diff
@@ -23,6 +23,11 @@
 namespace cuspatial {
 
 namespace detail {
+
+/**
+ * @internal
+ * @brief Helper to convert a tuple of elements into a `vec_2d`
```

Comment:
This question is out of scope for this PR, but why a tuple and not a pair? Issues with thrust::pair?

Reply:
why is a pair preferable to a tuple?

Reply:
Preferable is both subjective and circumstantial. As far as subjective arguments, on one hand we're making a 2d vector, which is exactly a pair of values, so it seems intuitive to use a pair. On the other hand I guess you could argue that you might want to generalize this to other dimensionality and returning a tuple across all dimensions would be more consistent. It's also convenient to use `pair.first/second` accessors rather than having to call a method.

As far as circumstantial arguments, the only real important thing that I can see is that in some situations you might care that pairs are standard layout while tuples are not. I don't think that really matters for us here.

I don't particularly think that we need to change this, just thought I'd check.